runtime/executor/src/main/java/org/apache/nemo/runtime/executor/data/partition/SerializedPartition.java

```diff
@@ -226,4 +236,11 @@ public int getLength() throws IOException {
   public boolean isOffheap() {
     return offheap;
   }
+
+  /**
+   * Releases the off-heap memory that this SerializedPartition holds.
+   */
+  public void release() {
+    bytesOutputStream.release();
```

Comment:
Shouldn't this be called upon commit()?

write(): writes to the stream
commit(): obtains the materialized data from the stream
getData(): returns the data

Given the above, I'd think that release() operates on the data, not the stream.

Reply:
Actually, `release()` operates on the *data*, although it seems like it works on the stream. The stream here uses a list of ByteBuffers wrapped as MemoryChunk, which is the data itself. The `dataList` in SerializedPartition shares the list of ByteBuffers of the output stream, so if this is called upon commit(), it will return the 'uncleared' MemoryChunks back to the pool, causing correctness problem. Maybe I could add some comments on this?

Reply:
I'll think more about this one.